src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs

```diff
@@ -823,25 +819,15 @@ internal EXPR BindToField(EXPR pOptionalObject, FieldWithType fwt, BindingFlag b
 
             EXPRFIELD pResult;
             {
-                bool isLValue = false;
-                if ((pOptionalObject != null && pOptionalObject.type.IsPointerType()) || objectIsLvalue(pOptionalObject))
-                {
-                    isLValue = true;
-                }
+                bool isLValue = pOptionalObject != null && pOptionalObject.type.IsPointerType() || objectIsLvalue(pOptionalObject);
```

Comment:
Please bring back the parenthesis around the && subexpression (left part of ||). It's easier to see && / || precedence IMO.
I would also love to see them around the != subexpr, but as above - that is not part of official coding style :(